src/metadata.rs

```diff
@@ -1,81 +1,431 @@
+use polars::{
+    frame::{explode, DataFrame},
+    lazy::{
+        dsl::{col, Expr},
+        frame::{IntoLazy, LazyFrame, ScanArgsParquet},
+    },
+    prelude::{lit, JoinArgs, JoinType, NamedFrom, UnionArgs},
+    series::Series,
+};
+use regex::Regex;
 use serde::{Deserialize, Serialize};
-use typify::import_types;
+use std::{collections::HashMap, default::Default};
 
 use crate::parquet::MetricRequest;
 
-import_types!("schema/popgetter_0.1.0.json");
+/// This struct contains the base url and names of
+/// the files that contain the metadata. It has a
+/// default impl which give the version that we will
+/// normally use but this allows us to customise it
+/// if we need to.
+pub struct CountryMetadataPaths {
+    base_url: String,
+    geometry: String,
+    metrics: String,
+    country: String,
+    source_data: String,
+    data_publishers: String,
+}
+
+/// Represents a way of refering to a metric id
+/// can be converted into a polars expression for
+/// selection
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub enum MetricId {
+    /// Hxl (Humanitarian Exchange Language) tag
+    Hxl(String),
+    /// Internal UUID
+    Id(String),
+    /// Human Readable name
+    CommonName(String),
+}
+
+impl MetricId {
+    /// Returns the column in the metadata that this id type corrispondes to
+    pub fn to_col_name(&self) -> String {
+        match self {
+            MetricId::Hxl(_) => "hxl_tag".into(),
+            MetricId::Id(_) => "id".into(),
+            MetricId::CommonName(_) => "human_readable_name".into(),
+        }
+    }
+    /// Return a string representing the textual content of the ID
+    pub fn to_query_string(&self) -> &str {
+        match self {
+            MetricId::Hxl(s) => s,
+            MetricId::Id(s) => s,
+            MetricId::CommonName(s) => s,
+        }
+    }
 
-impl SourceDataRelease{
-    pub fn get_metric_details(&self,metric_id: &str)->Option<&MetricMetadata>{
-        self.available_metrics.iter().find(
-            |m| m.source_metric_id == metric_id
-        ) 
+    /// Generate a polars Expr that will do
+    /// an exact match on the MetricId
+    pub fn to_polars_expr(&self) -> Expr {
+        col(&self.to_col_name()).eq(self.to_query_string())
+    }
+
+    /// Generate a polars Expr that will generate
+    /// a fuzzy search for the content of the Id
+    pub fn to_fuzzy_polars_expr(&self) -> Expr {
+        col(&self.to_col_name())
+            .str()
+            .contains(lit(self.to_query_string()), false)
     }
 }
 
-// TODO we might want to just pass the MetricMetaData rather than
-// having to 
-impl From<MetricMetadata> for MetricRequest{
-    fn from(value: MetricMetadata) -> Self {
-        MetricRequest { 
-            column: value.parquet_column_name.clone(),
-            file: value.metric_parquet_file_url.clone().unwrap()
-         }
+impl Into<Expr> for MetricId {
+    fn into(self) -> Expr {
+        self.to_polars_expr()
     }
 }
```

Comment:
Clippy [suggests](https://rust-lang.github.io/rust-clippy/master/index.html#/from_over_into) `impl From` over `impl Into`: 
```suggestion
impl From<MetricId> for Expr {
    fn from(value: MetricId) -> Self {
        value.to_polars_expr()
    }
}
```